third_party/blink/renderer/platform/loader/fetch/resource.h

```diff
@@ -489,7 +494,9 @@ class PLATFORM_EXPORT Resource : public GarbageCollected<Resource>,
   // Returns the memory dump name used for tracing. See Resource::OnMemoryDump.
   String GetMemoryDumpName() const;
 
-  const HeapHashCountedSet<WeakMember<ResourceClient>>& Clients() const {
+  const HeapHashCountedSet<WeakMember<ResourceClient>,
+                           WTF::MemberHashRecordReplayId<ResourceClient>>&
```

Comment:
It would be really nice if we could just change the default hash for the ResourceClient etc. so that changes like this aren't necessary.  Can we figure out an idiomatic way of doing this, e.g. with template specializations?

Reply:
We all agree with that! As I said [here](https://linear.app/replay/issue/RUN-1724#comment-f7885b73), the proper solution (`RUN-1665`) is in the pipeline. However, we are increasing priority on getting this out the door to increase chances of hitting a deadline.
The good thing is that undoing this is straight-forward, since all we need to do is just remove all occurrences of `MemberHashRecordReplayId`.

Reply:
I added a "clean up issue" here: https://linear.app/replay/issue/RUN-1741/deterministic-set-map-cleanup